Core/include/Acts/Seeding/CandidatesForMiddleSp.hpp

```diff
@@ -0,0 +1,187 @@
+  void pop(std::vector<std::size_t>& storage, std::size_t& current_size);
+
+  /// @brief Return the weight for a candidate
+  /// @param storage The collection in which the element is stored
+  /// @param n Index of the element in the collection
+  /// @returns The weight of the candidate
+  float weight(const std::vector<std::size_t>& storage, std::size_t n) const;
+
+  /// @brief Move an element up in the min heap tree. The function checks whether the element's
+  /// weight is lower of it's parent's weight. If so, it swaps them. Reiterate
+  /// the process untill the element is in the correct position on the tree
+  /// @param storage The collection
+  /// @param n The index of the element to place in the correct position
+  void bubbleup(std::vector<std::size_t>& storage, std::size_t n);
+
+  /// @brief Move an element down in the min heap tree. The function checks whether the elements's
+  /// weight is lower of it's child's weights. If so, it swaps the element with
+  /// the child with the lowest weight. Reiterate the process untill the element
+  /// is in the correct position on the tree
+  /// @param storage The collection
+  /// @param n The index of the element to place in the correct position
+  /// @param actual_size The current number of elements stored in the collection
+  void bubbledw(std::vector<std::size_t>& storage, std::size_t n,
+                std::size_t actual_size);
+
+  /// @brief Sdding a new triplet candidate to the collection. The function is called after the candidate has satisfied
+  /// all the selection criteria
+  /// @param storage The collection
+  /// @param n Current number of stored elements in the collection
+  /// @param n_max The maximum number of elements that can be stored in the collection
+  /// @param element The element that must be added to the collection
+  void addToCollection(std::vector<std::size_t>& storage, std::size_t& n,
+                       const std::size_t& n_max, value_type&& element);
+
+ private:
+  // sizes
+  // m_max_size_* is the maximum size of the storage collection. These values
+  // are set by the user once
+  std::size_t m_max_size_high{0};
+  std::size_t m_max_size_low{0};
+  // m_n_* is the corrent size of the storage collection [0, m_max_size_*).
+  // These values are set internally by the class
+  std::size_t m_n_high{0};
+  std::size_t m_n_low{0};
+
+  // storage contains the collection of the candidates
+  std::vector<value_type> m_storage{};
+
+  // The following vectors store indexes to elements in the storage
+  // They are sorted as a min heap tree, in which
+  // Each node is lower then its childs
+  // Thus, it is guaranteed that the lower elements is at the front
+  // Sorting criteria is the seed quality
+  //
+  // This is in effect faster sorted container - implementation with std::set
+  // and std::priority_queue  were tried and found slower.
+
+  // storage for candidates with high quality
+  std::vector<std::size_t> m_storage_high{};
```

Comment:
So these are indices in fact, not the storage, right? What do you think about naming these variables accordingly?

Reply:
changed to `m_indices_high` and `m_indices_low`